packages/core/lib/console.js

```diff
@@ -267,20 +267,31 @@ class Console extends EventEmitter {
       return callback(error);
     }
 
-    // Ensure our script returns a promise whether we're using an
-    // async function or not. If our script is an async function,
-    // this will ensure the console waits until that await is finished.
-    Promise.resolve(runScript(script))
-      .then(value => {
-        // If there's an assignment to run, run that.
-        if (assignment) return runScript(vm.createScript(assignment));
-        return value;
-      })
-      .then(value => {
-        // All good? Return the value (e.g., eval'd script or assignment)
-        callback(null, value);
-      })
-      .catch(callback);
+    const runScriptResult = runScript(script);
+
+    if (runScriptResult instanceof Promise) {
```

Comment:
So adding this `if` statement actually breaks some functionality here. Can you remove it?

You can confirm this by running `truffle develop` in a project directory, running migrations, and then doing something like 
```
const instance = await MyContract.deployed();
```

I think the script is run so quickly that the code takes the `else` and not the `if`. This causes `instance` to not get an assignment.

Reply:
Oh maybe it is not that it runs so quickly but rather Truffle supports `await` in the console and it finishes and resolves before checking the conditional. We'll need to implement this differently.

Reply:
yeah. It seems to be caused by await special handling. I will try to fix it